### PR TITLE
Bitwarden import

### LINKED ISSUE #222: Support import from Bitwarden
Here's what the exported data looks like:

```json
{
  "encrypted": false,
  "folders": [],
  "items": [
    {
      "passwordHistory": null,
      "revisionDate": "2023-11-08T06:14:57.798Z",
      "creationDate": "2023-11-08T06:14:57.798Z",
      "deletedDate": null,
      "id": "be3f638e-e389-4525-9a07-b0b40066fcf8",
      "organizationId": null,
      "folderId": null,
      "type": 1,
      "reprompt": 0,
      "name": "Github",
      "notes": null,
      "favorite": false,
      "login": {
        "fido2Credentials": [],
        "username": "test@ente.io",
        "password": "password",
        "totp": "2V5GHKTPKPZDZNXUDMJ6GBFL55RJBJUE"
      },
      "collectionIds": null
    },
    {
      "passwordHistory": null,
      "revisionDate": "2023-11-08T06:14:17.821Z",
      "creationDate": "2023-11-08T06:14:17.821Z",
      "deletedDate": null,
      "id": "e68bc63f-2374-453c-a259-b0b40066ce3c",
      "organizationId": null,
      "folderId": null,
      "type": 1,
      "reprompt": 0,
      "name": "Cloudflare",
      "notes": null,
      "favorite": false,
      "login": {
        "fido2Credentials": [],
        "username": "test@ente.io",
        "password": "password",
        "totp": null
      },
      "collectionIds": null
    }
  ]
}
```

We should import `name`, `login.username` and `login.totp` from `items` as `issuer`, `account` and `secret` where `login.totp` is not `null`.

### PR DESCRIPTION
**Pull request by i-aiymen**
## Description

Added: Import from bitwarden functionality
Closes #222 

## Type of Change

- [ ] 🖼️ New icon
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by i-aiymen** [comment]
@vishnukvmd @ua741

File: lib/l10n/arb/app_en.arb [hunk 1]
```diff
@@ -86,6 +86,7 @@
   "importSelectJsonFile": "Select JSON file",
   "importEnteEncGuide": "Select the encrypted JSON file exported from ente",
   "importRaivoGuide": "Use the \"Export OTPs to Zip archive\" option in Raivo's Settings.\n\nExtract the zip file and import the JSON file.",
+  "importBitwardenGuide": "Use the \"Export\" option in Bitwarden Settings.\n\nExtract the zip file and import the JSON file.",
```
Comment (vishnukvmd) [inline lib/l10n/arb/app_en.arb]: "Use the \"Export vault\" option within Bitwarden Tools and import the unencrypted JSON file.";

File: lib/ui/settings/data/import/bitwarden_import.dart [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        buttonSize: ButtonSize.large,
+        isInAlert: true,
+        buttonAction: ButtonAction.second,
+      ),
+    ],
+  );
+  if (result?.action != null && result!.action != ButtonAction.cancel) {
+    if (result.action == ButtonAction.first) {
+      await _pickBitwardenJsonFile(context);
+    } else {}
```
Comment (vishnukvmd) [inline lib/ui/settings/data/import/bitwarden_import.dart]: Can drop the empty `else` block

File: lib/ui/settings/data/import/bitwarden_import.dart [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    );
+  }
+}
+
+Future<int?> _processBitwardenExportFile(
+  BuildContext context,
+  String path,
+) async {
+  File file = File(path);
+  if (path.endsWith('.zip')) {
```
Comment (vishnukvmd) [inline lib/ui/settings/data/import/bitwarden_import.dart]: Bitwarden exports it directly as a `.json` file, this check is unnecessary

File: lib/ui/settings/data/import/bitwarden_import.dart [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        ),
+      );
+    }
+  }
+
+  for (final code in parsedCodes) {
+    await CodeStore.instance.addCode(code, shouldSync: false);
+  }
+  unawaited(AuthenticatorService.instance.onlineSync());
+  int count = parsedCodes.length;
```
Comment (vishnukvmd) [inline lib/ui/settings/data/import/bitwarden_import.dart]: No point extracting this to a variable?

**Review by vishnukvmd — COMMENTED** [review]
Could you confirm that the import is working for the sample data provided in the issue description @ #222?

File: lib/l10n/arb/app_en.arb (hunk 1 above)
Comment (i-aiymen) [reply lib/l10n/arb/app_en.arb]: ok

File: lib/ui/settings/data/import/bitwarden_import.dart (hunk 2 above)
Comment (i-aiymen) [reply lib/ui/settings/data/import/bitwarden_import.dart]: ok

File: lib/ui/settings/data/import/bitwarden_import.dart (hunk 3 above)
Comment (i-aiymen) [reply lib/ui/settings/data/import/bitwarden_import.dart]: ok

File: lib/ui/settings/data/import/bitwarden_import.dart (hunk 4 above)
Comment (i-aiymen) [reply lib/ui/settings/data/import/bitwarden_import.dart]: ok

**Comment by i-aiymen** [comment]
@vishnukvmd resolved the fixes

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**